### PR TITLE
[cpp] receiver queue size config acorss partitions in multi-topics-consumer

### PR DESCRIPTION
**Pull request by jiazhai**
### Motivation

In PR #1123, we add a configuration of  max producer/consumer queue size enforced across partitions
This is an catch up work to support in multi-topics-consumer.

### Modifications

Apply total limit of receiver queue size across partitions

### CODE REVIEW

**Comment by sijie** [comment]
looks good to me. are you planning to add the config to python as well?

File: pulsar-client/src/main/java/org/apache/pulsar/client/impl/MultiTopicsConsumerImpl.java [hunk 1]
```diff
@@ -665,24 +665,29 @@ private boolean topicNameValid(String topicName) {
...
-            log.debug("Subscribe to topic {} metadata.partitions: {}", topicName, partitionNumber);
+            log.debug("Subscribe to topic {} metadata.partitions: {}", topicName, numPartitions);
         }
 
         List<CompletableFuture<Consumer<T>>> futureList;
 
-        if (partitionNumber > 1) {
-            this.topics.putIfAbsent(topicName, partitionNumber);
-            allTopicPartitionsNumber.addAndGet(partitionNumber);
+        if (numPartitions > 1) {
```
Comment (merlimat) [inline pulsar-client/src/main/java/org/apache/pulsar/client/impl/MultiTopicsConsumerImpl.java]: Why do we need  this change in Java client as well?

Comment (jiazhai) [reply pulsar-client/src/main/java/org/apache/pulsar/client/impl/MultiTopicsConsumerImpl.java]: Thanks. PartitionConsumer is using MultiTopicsConsumer now, and seems the configuration of queue size  across partitions was removed. this is a work to add it back.

**Comment by jiazhai** [comment]
@sijie added python config

**Comment by jiazhai** [comment]
rerun integration tests

**Comment by sijie** [comment]
@merlimat please review this PR again

**Comment by sijie** [comment]
merged as 
bcb3d3f in master

**Pull request merged**